src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random random = new Random();
+
+    public static Figure getRandomFigure() {
+        int choice = random.nextInt(5);
+        ColorSupplier colorSupplier = new ColorSupplier();
```

Comment:
ColorSupplier object creates every time you call getRandomFigure. Think of approach, when you create an object of ColorSupplier only once for this class